modules/core/src/main/java/org/apache/ignite/lang/IgniteUuid.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.lang;
+
+import java.io.Serializable;
+import java.util.UUID;
+
+/**
+ * This is a faster performing version of {@link UUID}. On basic tests this version is at least
+ * 10x time faster for ID creation. It uses extra memory for 8-byte counter additionally to
+ * internal UUID.
+ */
+public final class IgniteUuid implements Comparable<IgniteUuid>, Cloneable, Serializable {
```

Comment:
Let's describe IgniteUuid natural order here.

Reply:
Not a scope. Could you please describe it during storage related work?